huaweicloud/data_source_huaweicloud_compute_instance.go

```diff
@@ -165,113 +168,63 @@ func DataSourceComputeInstance() *schema.Resource {
 	}
 }
 
-func dataSourceComputeInstanceRead(d *schema.ResourceData, meta interface{}) error {
-	config := meta.(*config.Config)
-	ecsClient, err := config.ComputeV1Client(GetRegion(d, config))
-	if err != nil {
-		return fmtp.Errorf("Error creating HuaweiCloud ECS client: %s", err)
-	}
-
-	epsID := GetEnterpriseProjectID(d, config)
-	listOpts := &cloudservers.ListOpts{
+func buildEcsListOpts(d *schema.ResourceData, conf *config.Config) *cloudservers.ListOpts {
+	result := cloudservers.ListOpts{
+		Limit:               100,
+		EnterpriseProjectID: GetEnterpriseProjectID(d, conf),
 		Name:                d.Get("name").(string),
 		Flavor:              d.Get("flavor_id").(string),
+		Status:              d.Get("status").(string),
```

Comment:
status is an attribute, the value is always empty

Reply:
There are two kinds of updation:
1. Give the parameter `status` a `Optional` behavior.
2. Remove the `status` from ListOpts built method.
Maybe the first one is better.